cmd/finch/main_darwin.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build darwin
+
+package main
+
+import (
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/config"
+	"github.com/runfinch/finch/pkg/dependency"
+	"github.com/runfinch/finch/pkg/dependency/credhelper"
+	"github.com/runfinch/finch/pkg/dependency/vmnet"
+	"github.com/runfinch/finch/pkg/disk"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/fssh"
+	"github.com/runfinch/finch/pkg/path"
+	"github.com/runfinch/finch/pkg/system"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func virtualMachineCommands(
+	logger flog.Logger,
+	fp path.Finch,
+	lcc command.LimaCmdCreator,
+	ecc *command.ExecCmdCreator,
+	fs afero.Fs,
+	fc *config.Finch,
+) *cobra.Command {
+	optionalDepGroups := []*dependency.Group{
+		credhelper.NewDependencyGroup(ecc, fs, fp, logger, fc, system.NewStdLib().Env("USER"),
+			system.NewStdLib().Arch()),
+	}
+
+	optionalDepGroups = append(optionalDepGroups, vmnet.NewDependencyGroup(ecc, lcc, fs, fp, logger))
```

Comment:
Is it possible to only extract the depGroups into a new platform specific file? Seems like everything else is the same